perfmetrics/scripts/run_e2e_tests.sh

```diff
@@ -138,69 +148,146 @@ function run_parallel_tests() {
   return $exit_code
 }
 
-sudo apt-get update
-# Upgrade gcloud version.
-# Kokoro machine's outdated gcloud version prevents the use of the "managed-folders" feature.
-upgrade_gcloud_version
-install_packages
-
-# Test setup
-# Create Bucket for non parallel e2e tests
-# The bucket prefix for the random string
-bucket_prefix="gcsfuse-non-parallel-e2e-tests-group-1-"
-bucket_name_non_parallel_group_1=$(create_bucket $bucket_prefix)
-echo "Bucket name for non parallel tests group - 1: "$bucket_name_non_parallel_group_1
-
-# Test setup
-# Create Bucket for non parallel e2e tests
-# The bucket prefix for the random string
-bucket_prefix="gcsfuse-non-parallel-e2e-tests-group-2-"
-bucket_name_non_parallel_group_2=$(create_bucket $bucket_prefix)
-echo "Bucket name for non parallel tests group - 2 : "$bucket_name_non_parallel_group_2
-
-# Create Bucket for parallel e2e tests
-# The bucket prefix for the random string
-bucket_prefix="gcsfuse-parallel-e2e-tests-"
-bucket_name_parallel=$(create_bucket $bucket_prefix)
-echo "Bucket name for parallel tests: "$bucket_name_parallel
-
-# Run tests
-set +e
-echo "Running parallel tests..."
-# Run parallel tests
-run_parallel_tests TEST_DIR_PARALLEL $bucket_name_parallel &
-parallel_tests_pid=$!
-
-# Run non parallel tests
-echo "Running non parallel tests group-1..."
-run_non_parallel_tests TEST_DIR_NON_PARALLEL_GROUP_1 $bucket_name_non_parallel_group_1 &
-non_parallel_tests_pid_group_1=$!
-echo "Running non parallel tests group-2..."
-run_non_parallel_tests TEST_DIR_NON_PARALLEL_GROUP_2 $bucket_name_non_parallel_group_2 &
-non_parallel_tests_pid_group_2=$!
-
-# Wait for all tests to complete.
-wait $parallel_tests_pid
-parallel_tests_exit_code=$?
-wait $non_parallel_tests_pid_group_1
-non_parallel_tests_exit_code_group_1=$?
-wait $non_parallel_tests_pid_group_2
-non_parallel_tests_exit_code_group_2=$?
-set -e
-
-# Cleanup
-# Delete bucket after testing.
-gcloud alpha storage rm --recursive gs://$bucket_name_parallel/
-gcloud alpha storage rm --recursive gs://$bucket_name_non_parallel_group_1/
-gcloud alpha storage rm --recursive gs://$bucket_name_non_parallel_group_2/
-
-# Removing bin file after testing.
-if [ $RUN_E2E_TESTS_ON_PACKAGE != true ];
-then
-  sudo rm /usr/local/bin/gcsfuse
-fi
-if [ $non_parallel_tests_exit_code_group_1 != 0 ] || [ $non_parallel_tests_exit_code_group_2 != 0 ] || [ $parallel_tests_exit_code != 0 ];
-then
-  echo "The tests failed."
-  exit 1
-fi
+function create_flat_buckets() {
+  # Test setup
+  # Create Bucket for non parallel e2e tests
+  # The bucket prefix for the random string
+  bucketPrefix="gcsfuse-non-parallel-e2e-tests-group-1-"
+  BUCKET_NAME_NON_PARALLEL_GROUP_1=$(create_bucket $bucketPrefix)
```

Comment:
this shouldn't be in 
(a) CAPS_LOCK because it's not global variable.
(b) It is set here and then used in different function.

Would suggest to just keep one function run_e2e_tests_for_flat_bucket and create buckets inside it.

Reply:
I think you would also need to call clean_up from inside this for the same reason.

Reply:
that ll make function very long , want to avoid that

Reply:
Ideally, the bucket names should be passed as argument between functions. However, I feel that will be overkill here. I would maintain consistency and use CAPS_LOCK for constants and allow the function to grow big. Wdyt ?